File: algorithms/2_medium/B/01441_build_an_array_with_stack_operations/simulation.py
class Solution:
    def buildArray(self, target: List[int], n: int) -> List[str]:
        target_len = len(target)
        i = 0
        num = 1
        res = []

        while i < target_len:
            res.append("Push")

            if num == target[i]:
                i += 1
            else:
                res.append("Pop")

            num += 1

        return res

File: algorithms/2_medium/B/01441_build_an_array_with_stack_operations/simulation.rs
impl Solution {
    pub fn build_array(target: Vec<i32>, n: i32) -> Vec<String> {
        let target_len = target.len();
        let mut target_index = 0_usize;
        let mut num = 1_i32;
        let mut res: Vec<String> = Vec::new();

        while target_index <  target_len{
            res.push("Push".to_string());

            if num == target[target_index] {
                target_index += 1;
            } else {
                res.push("Pop".to_string());
            }

            num += 1;
        }

        res
    }
}
